From 7a9108595de2a3c6612a3ec92e20b01b610864df Mon Sep 17 00:00:00 2001
From: Larry Li <b20787@freescale.com>
Date: Mon, 25 Jun 2012 16:09:23 +0800
Subject: ENGR00214791-2 [MX6] GalCore gets baseAddress parameter for kernel

Galcore reads baseAddress parameter from GPU resource set by kernel

Signed-off-by: Larry Li <b20787@freescale.com>
Acked-by: Lily Zhang
---
 .../hal/os/linux/kernel/gc_hal_kernel_driver.c     |    4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/mxc/gpu-viv/hal/os/linux/kernel/gc_hal_kernel_driver.c b/drivers/mxc/gpu-viv/hal/os/linux/kernel/gc_hal_kernel_driver.c
index 1786d09..24775ad 100644
--- a/drivers/mxc/gpu-viv/hal/os/linux/kernel/gc_hal_kernel_driver.c
+++ b/drivers/mxc/gpu-viv/hal/os/linux/kernel/gc_hal_kernel_driver.c
@@ -929,6 +929,10 @@ static int __devinit gpu_probe(struct platform_device *pdev)
 
     gcmkHEADER();
 
+    res = platform_get_resource_byname(pdev, IORESOURCE_MEM, "phys_baseaddr");
+    if (res)
+        baseAddress = res->start;
+
     res = platform_get_resource_byname(pdev, IORESOURCE_IRQ, "irq_3d");
     if (res)
         irqLine = res->start;
-- 
1.7.9.5

